### PR TITLE
View part 07

### PR DESCRIPTION
**Pull request by at-bienle**
help me :v i finished view!

### CODE REVIEW

File: ViewPart07/ViewPart07/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+    var window: UIWindow?
+
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+
+        self.window = UIWindow(frame: UIScreen.main.bounds)
+        self.window?.backgroundColor = UIColor.white
+        self.window?.makeKeyAndVisible()
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/AppDelegate.swift]: remove `self` please

File: ViewPart07/ViewPart07/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+    }
+
+    func applicationDidBecomeActive(_ application: UIApplication) {
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    }
+
+    func applicationWillTerminate(_ application: UIApplication) {
+        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    }
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/AppDelegate.swift]: and please remove unused code

File: ViewPart07/ViewPart07/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+//  Created by AST on 6/14/17.
+//  Copyright © 2017 AST. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var sliderView: UIView!
+    @IBOutlet weak var button: UIButton!
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/ViewController.swift]: `button` name is not clear, pls change it

**Comment by at-hieutran** [comment]
@at-bienle Fix conflict, please.

File: ViewPart07/ViewPart07/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                button.center.y = loc.y >= sliderView.frame.origin.y + sliderView.bounds.width / 2 ? sliderView.frame.origin.y + sliderView.frame.height : sliderView.frame.origin.y
+                let x: Int = loc.y >= sliderView.frame.origin.y + sliderView.bounds.width / 2 ? 0 : 100
+                button.setTitle(String(x), for: .normal)
+            }
+        }
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        guard let loc = touches.first?.location(in: view) else { return }
+        if (loc.x < button.center.x + button.frame.width / 2) && (loc.x > button.center.x - button.frame.width / 2) && (loc.y < button.center.y + button.frame.height / 2) && (loc.y > button.center.y - button.frame.width / 2)
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/ViewController.swift]: button.center.x + button.frame.width / 2 = button.frame.maxX
button.center.x - button.frame.width / 2 = button.frame.minX
button.center.y + button.frame.height / 2 = button.frame.maxY
button.center.y - button.frame.width / 2 = button.frame.minY

File: ViewPart07/ViewPart07/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var sliderView: UIView!
+    @IBOutlet weak var button: UIButton!
+    @IBOutlet weak var blueView: UIView!
+    
+    var isTouchOnButton = false
+
+    override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/ViewController.swift]: miss `super`

File: ViewPart07/ViewPart07/ViewController.swift [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destinationViewController.
+        // Pass the selected object to the new view controller.
+    }
+    */
```
Comment (at-truongnguyen) [inline ViewPart07/ViewPart07/ViewController.swift]: remove unnecessary code

**Comment by at-bienle** [comment]
ANH OI, REVIEW EM VOI 🍡

**Comment by at-hieutran** [comment]
OK baby.

**Pull request merged**